toolchain/driver/clang_runner.cpp

```diff
@@ -113,9 +124,15 @@ auto ClangRunner::Run(llvm::ArrayRef<llvm::StringRef> args) -> bool {
   driver.Dir = installation_->llvm_install_bin();
   CARBON_VLOG("Setting bin directory to: {0}\n", driver.Dir);
 
-  // TODO: Directly run in-process rather than using a subprocess. This is both
-  // more efficient and makes debugging (much) easier. Needs code like:
-  // driver.CC1Main = [](llvm::SmallVectorImpl<const char*>& argv) {};
+  // When there's only one command being run, this will run it in-process.
+  // However, a `clang` invocation may cause multiple `cc1` invocations, which
+  // still subprocess.
```

Comment:
This makes sense to me.

I have an idea of another step we may want to eventually try to take here, but emphasis on "may" (not at all sure) and "eventually". =] Suggesting as a TODO in case you want to capture it and/or tweak the wording in the code. But also happy to just leave for discussion or the future.

```suggestion
  // When there's only one command being run, this will run it in-process.
  // However, a `clang` invocation may cause multiple `cc1` invocations, which
  // still subprocess.
  //
  // TODO: It would be nice to find a way to set up the driver's understanding
  // of the executable name in a way that causes the multiple `cc1` invocations
  // to actually result in `carbon clang -- ...` invocations (even if as
  // subprocesses). This may dovetail with having symlinks that redirect to a
  // busybox of LLD as well, and having even the subprocesses consistently run
  // the Carbon install toolchain and not a system toolchain whenever possible.
```

Maybe this is already part of what you're thinking about with busyboxing the linker too, feel free to skip the TODO if already part of your larger plans.

Reply:
Added. Also yes, that's what I've been working towards with busyboxing and why I was examining the subprocess behavior of ClangRunner.